airflow/auth/managers/base_auth_manager.py

```diff
@@ -443,3 +430,19 @@ def security_manager(self) -> AirflowSecurityManagerV2:
         from airflow.www.security_manager import AirflowSecurityManagerV2
 
         return AirflowSecurityManagerV2(self.appbuilder)
+
+    @staticmethod
+    def get_cli_commands() -> list[CLICommand]:
+        """
+        Vends CLI commands to be included in Airflow CLI.
+
+        Override this method to expose commands via Airflow CLI to manage this auth manager.
+        """
+        return []
+
+    def get_api_endpoints(self) -> None | Blueprint:
+        """Return API endpoint(s) definition for the auth manager."""
+        return None
+
+    def register_views(self) -> None:
```

Comment:
This method does not carry any parameter and has no return. The only way to "register" something is - as seen in the AWS auth mgr - to use the FAB appbuilder class from the global app variable.

If we do it like this, we can not deprecate and make FAB optional. Is there any option not adding a new binding to FAB when we add the UI registration? Bare Flask endpoints/routes and menu names?

Reply:
FYI @bbovenzi or @pierrejeambrun this smells a bit like the idea of AIP-68 but for another purpose - do you have an idea how to expose "views" w/o destroying the future UI concept in React? (w/o having the new plugin mechanism in place today and and not adding a new binding to FAB?

Reply:
As mentioned [here](https://github.com/apache/airflow/pull/41777#issuecomment-2316299463), this is not in the scope of this PR